cf/forest-snapshot-listing/src/worker.ts

```diff
@@ -0,0 +1,83 @@
+const BASE_URL = 'https://forest-archive.chainsafe.dev';
+
+async function do_listing(env, prefix) {
+	const options = {
+		limit: 500,
+		prefix: prefix,
+	};
+
+	const listed = await env.FOREST_ARCHIVE.list(options);
+	let truncated = listed.truncated;
+	let cursor = truncated ? listed.cursor : undefined;
+
+	// use the truncated property to check if there are more
+	// objects to be returned
+	while (truncated) {
+		const next = await env.FOREST_ARCHIVE.list({
+			...options,
+			cursor: cursor,
+		});
+		listed.objects.push(...next.objects);
+
+		truncated = next.truncated;
+		cursor = next.cursor;
+	}
+
+	let html = `<!DOCTYPE html>
+		<body>
+		  <h1>Forest Archive</h1>
+      <ul>`;
+	for (const obj of listed.objects) {
+		html += `<li><a href="${BASE_URL}/${obj.key}">${obj.key}</a></li>\n`;
+	}
+	html += `</ul></body>`;
+
+	return new Response(html, {
+		headers: {
+			'content-type': 'text/html;charset=UTF-8',
+		},
+	});
+}
+
+export default {
+	async fetch(request, env) {
+		const url = new URL(request.url);
+		const { pathname } = url;
+
+		switch (pathname) {
+			case '/list/': {
```

Comment:
make it more user-who-forgots-slashes-friendly
```suggestion
			case '/list':
			case '/list/': {
```